src/registrar/views/domain.py

```diff
@@ -644,21 +643,20 @@ def _domain_abs_url(self):
         """Get an absolute URL for this domain."""
         return self.request.build_absolute_uri(reverse("domain", kwargs={"pk": self.object.id}))
 
-    def _send_domain_invitation_email(self, email: str, add_success=True):
+    def _send_domain_invitation_email(self, email: str, requester: User, add_success=True):
         """Performs the sending of the domain invitation email,
         does not make a domain information object
         email: string- email to send to
         add_success: bool- default True indicates:
           adding a success message to the view if the email sending succeeds"""
-        # created a new invitation in the database, so send an email
-        domainInfoResults = DomainInformation.objects.filter(domain=self.object)
-        domainInfo = domainInfoResults.first()
-        first = ""
-        last = ""
-        if domainInfo is not None:
-            first = domainInfo.creator.first_name
-            last = domainInfo.creator.last_name
-        full_name = f"{first} {last}"
+
+        # Check if the email requester has a valid email address
+        if requester.email is not None and requester.email.strip() != "":
+            requester_email = requester.email
+        else:
+            # This edgecase would be unusual if encountered. We don't want to handle this here. Rather, we would
```

Comment:
**blocking** I disagree. with raising a valueError that produces a 500 server error which we should avoid in favor of a human readable message. I would instead look at how we handle the emailSendingError  in the below Except. There we add a message to the user. I think I would do this with messages.error  and be sure to log this with logger.info with something identifiable so we can find which user it is that is producing this issue (first/last name or UUID would be reasonable here).